featurebyte/service/feature_manager.py

```diff
@@ -259,22 +261,37 @@ async def online_disable(self, feature_spec: OnlineFeatureSpec) -> None:
         feature_spec: OnlineFeatureSpec
             input feature instance
         """
-        # delete records from tile-feature mapping table
-        for agg_id in feature_spec.aggregation_ids:
-            for query in feature_spec.precompute_queries:
-                try:
-                    await self.online_store_compute_query_service.delete_by_result_name(
-                        result_name=query.result_name
-                    )
-                except DocumentNotFoundError:
-                    # TODO: this is caused by an existing bug (https://featurebyte.atlassian.net/issues/DEV-1845)
-                    pass
-            logger.debug(f"Done delete online_store_mapping for {agg_id}")
+        await self.remove_online_store_compute_queries(feature_spec)
 
         # disable tile scheduled jobs
         for tile_spec in feature_spec.feature.tile_specs:
             await self.tile_manager_service.remove_tile_jobs(tile_spec)
 
+    async def remove_online_store_compute_queries(self, feature_spec: OnlineFeatureSpec) -> None:
+        """
+        Update the list of currently active online store compute queries
+
+        Parameters
+        ----------
+        feature_spec: OnlineFeatureSpec
+            Specification of the feature that is currently being online disabled
+        """
+        aggregation_result_names = {query.result_name for query in feature_spec.precompute_queries}
+        query_filter = {
+            "online_enabled": True,
+            "aggregation_result_names": {
+                "$in": list(aggregation_result_names),
+            },
```

Comment:
We may need update `FeatureModel` by including `aggregation_result_names` in the mongo indices.

Reply:
Seems like it is included in the model already.